dir.props

```diff
@@ -56,22 +56,16 @@
   <!-- Provides package dependency version properties and verification/auto-upgrade configuration -->
   <Import Project="$(MSBuildThisFileDirectory)dependencies.props" />
 
-  <ItemGroup>
-    <DotnetSourceList Include="$(DotNetSourceBuildRestoreSources)" />
-  </ItemGroup>
-
   <!-- list of nuget package sources passed to nuget restore -->
-  <ItemGroup Condition="'$(DotNetBuildOffline)' != 'true'">
-    <!-- Example to consume local CoreCLR package:
-         /p:OverridePackageSource=C:\coreclr\bin\Product\Windows_NT.x64.Debug\.nuget\pkg
-    -->
-    <DotnetSourceList Include="$(OverridePackageSource)" />
-    <!-- Need to escape double forward slash (%2F) or MSBuild will normalize to one slash on Unix. -->
-    <!-- Including buildtools to pull in TestSuite packages and repackaged xunit dependencies-->
-    <DotnetSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-buildtools/api/v3/index.json" />
-    <DotnetSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-core/api/v3/index.json" />
-    <DotnetSourceList Include="https:%2F%2Fapi.nuget.org/v3/index.json" />
-  </ItemGroup>
+  <PropertyGroup>
+    <RestoreSources Condition="'$(DotNetBuildOffline)' != 'true'">
+      $(RestoreSources);
+      $(OverridePackageSource);
+      https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json;
+      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json;
+      https://api.nuget.org/v3/index.json
+    </RestoreSources>
```

Comment:
@dagood have a look at https://github.com/dotnet/corefx/commit/436eda49c4e7781b1115f6ec38ffa61211b60ed4 as I suspect your changes will need to account for the nuget bug as well. You may need to actually make an update to buildtools because of it.

Reply:
Agreed this is affected, I'll make the order change. BuildTools shouldn't need a change: the only order enforcement is that all the sources in the restore source props file are adjacent, which I don't think is a problem now.

In theory, a Repo API caller is allowed to pass in a restore source props that has sources in "local, remote" order, causing that bug. We *could* have BuildTools detect that and sort the values in the workaround way, but that doesn't sound worth the cost to me. I think it's much more worthwhile to change the repos so that `dotnet restore` picks up `RestoreSources` properties "naturally" through the msbuild project structure rather than using `--source a --source b`.